src/modules/vtol_att_control/vtol_att_control_main.cpp

```diff
@@ -671,105 +651,74 @@ void VtolAttitudeControl::task_main()
-			_vtol_vehicle_status.in_transition_to_fw = (_vtol_type->get_mode() == TRANSITION_TO_FW);
-
 			_vtol_type->update_transition_state();
 			fill_mc_att_rates_sp();
 		}
 
 		_vtol_type->fill_actuator_outputs();
 
-		/* Only publish if the proper mode(s) are enabled */
-		if (_v_control_mode.flag_control_attitude_enabled ||
-		    _v_control_mode.flag_control_rates_enabled ||
-		    _v_control_mode.flag_control_manual_enabled) {
+		if (_v_att_sp_pub != nullptr) {
+			orb_publish(ORB_ID(vehicle_attitude_setpoint), _v_att_sp_pub, &_v_att_sp);
 
-			if (_v_att_sp_pub != nullptr) {
-				/* publish the attitude setpoint */
-				orb_publish(ORB_ID(vehicle_attitude_setpoint), _v_att_sp_pub, &_v_att_sp);
+		} else {
+			_v_att_sp_pub = orb_advertise(ORB_ID(vehicle_attitude_setpoint), &_v_att_sp);
+		}
 
-			} else {
-				/* advertise and publish */
-				_v_att_sp_pub = orb_advertise(ORB_ID(vehicle_attitude_setpoint), &_v_att_sp);
-			}
+		if (_actuators_0_pub != nullptr) {
+			orb_publish(ORB_ID(actuator_controls_0), _actuators_0_pub, &_actuators_out_0);
 
-			if (_actuators_0_pub != nullptr) {
-				orb_publish(ORB_ID(actuator_controls_0), _actuators_0_pub, &_actuators_out_0);
+		} else {
+			_actuators_0_pub = orb_advertise(ORB_ID(actuator_controls_0), &_actuators_out_0);
+		}
 
-			} else {
-				_actuators_0_pub = orb_advertise(ORB_ID(actuator_controls_0), &_actuators_out_0);
-			}
+		if (_actuators_1_pub != nullptr) {
+			orb_publish(ORB_ID(actuator_controls_1), _actuators_1_pub, &_actuators_out_1);
 
-			if (_actuators_1_pub != nullptr) {
-				orb_publish(ORB_ID(actuator_controls_1), _actuators_1_pub, &_actuators_out_1);
+		} else {
+			_actuators_1_pub = orb_advertise(ORB_ID(actuator_controls_1), &_actuators_out_1);
+		}
 
-			} else {
-				_actuators_1_pub = orb_advertise(ORB_ID(actuator_controls_1), &_actuators_out_1);
-			}
+		bool publish_status = false;
+
+		if (hrt_elapsed_time(&_vtol_vehicle_status.timestamp) > 1_s) {
+			publish_status = true;
 		}
 
-		/*Advertise/Publish vtol vehicle status*/
-		_vtol_vehicle_status.timestamp = hrt_absolute_time();
+		if (_vtol_vehicle_status.vtol_state != _vtol_type->get_mode()) {
+			publish_status = true;
+		}
 
-		if (_vtol_vehicle_status_pub != nullptr) {
-			orb_publish(ORB_ID(vtol_vehicle_status), _vtol_vehicle_status_pub, &_vtol_vehicle_status);
+		if (_vtol_vehicle_status.vtol_transition_failsafe != _abort_front_transition) {
+			publish_status = true;
+		}
 
-		} else {
-			_vtol_vehicle_status_pub = orb_advertise(ORB_ID(vtol_vehicle_status), &_vtol_vehicle_status);
+		if (publish_status) {
+			_vtol_vehicle_status.timestamp = hrt_absolute_time();
+			_vtol_vehicle_status.vtol_state = _vtol_type->get_mode();
+
+			if (_vtol_vehicle_status_pub != nullptr) {
+				orb_publish(ORB_ID(vtol_vehicle_status), _vtol_vehicle_status_pub, &_vtol_vehicle_status);
+
+			} else {
+				_vtol_vehicle_status_pub = orb_advertise(ORB_ID(vtol_vehicle_status), &_vtol_vehicle_status);
+			}
```

Comment:
I haven't reviewed this line-by-line (something for @RomanBapst) but the general cleanup looks good and the restructuring looks consistent.